### PR TITLE
Migrate to unstable dependencies.

### PR DESCRIPTION
**Pull request by frankban**
This branch also includes:
- fix tests making use of `charmstoretesting`, which does not exist anymore;
- sort imports;
- fix httpbakery usage to use the v1 package;
- fix cleaning up of some tests making use of mongo sessions;
- update dependencies file.

Please note that this is proposed for merging into the `guibundles` feature branch:
we plan to freeze unstable packages before including this work in core trunk.

(Review request: http://reviews.vapour.ws/r/2491/)

### CODE REVIEW

File: cmd/juju/commands/deploy_test.go [hunk 1]
```diff
@@ -499,9 +500,21 @@ func (s *charmStoreSuite) SetUpTest(c *gc.C) {
...
 	// Set up the charm store testing server.
-	s.srv = charmstoretesting.OpenServer(c, s.Session, charmstore.ServerParams{
+	db := s.Session.DB("juju-testing")
+	params := charmstore.ServerParams{
+		AuthUsername:     "test-user",
+		AuthPassword:     "test-password",
 		IdentityLocation: s.discharger.Location(),
 		PublicKeyLocator: s.discharger,
+	}
+	handler, err := charmstore.NewServer(db, nil, "", params, charmstore.V4)
```
Comment (mattyw) [inline cmd/juju/commands/deploy_test.go]: Is charmstore.V4 right here or should it be V5?

File: cmd/juju/commands/register.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 
 	"github.com/juju/errors"
 	"github.com/juju/persistent-cookiejar"
-	"gopkg.in/macaroon-bakery.v0/httpbakery"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
```
Comment (mattyw) [inline cmd/juju/commands/register.go]: No action needed, just thanks for doing this

File: testcharms/charm.go [hunk 3]
```diff
@@ -6,8 +6,38 @@
...
 // Repo provides access to the test charm repository.
 var Repo = testing.NewRepo("charm-repo", "quantal")
+
+// UploadCharm uploads a charm using the given charm store client, and returns
+// the resulting charm URL and charm.
+func UploadCharm(c *gc.C, client *csclient.Client, url, name string) (*charm.URL, charm.Charm) {
+	id := charm.MustParseReference(url)
+	promulgatedRevision := -1
+	if id.User == "" {
+		id.User = "who"
```
Comment (mattyw) [inline testcharms/charm.go]: A comment stating why we do this would be useful

File: cmd/juju/commands/deploy_test.go (hunk 1 above)
Comment (frankban) [inline cmd/juju/commands/deploy_test.go]: It's V4. It's not the library version, it's the HTTP API version. I know, it's confusing.

File: testcharms/charm.go (hunk 3 above)
Comment (frankban) [inline testcharms/charm.go]: Done.

File: apiserver/testing/fakecharmstore.go [hunk 4]
```diff
@@ -45,32 +49,37 @@ func (s *CharmStoreSuite) SetUpTest(c *gc.C) {
...
+	handler, err := charmstore.NewServer(db, nil, "", params, charmstore.V4)
+	c.Assert(err, jc.ErrorIsNil)
+	s.handler = handler
+	s.Srv = httptest.NewServer(handler)
+	s.Client = csclient.New(csclient.Params{
+		URL:      s.Srv.URL,
+		User:     params.AuthUsername,
+		Password: params.AuthPassword,
 	})
+
```
Comment (urosj) [inline apiserver/testing/fakecharmstore.go]: This block of code is repeated a lot. We should make it easier to create without code repetition.

**Comment by urosj** [comment]
LGTM, but there's a comment about code repetition. It's nice to have such level of control over dependency between handlers, server and clients, but it would also be nice to just create all of those in one call.

File: apiserver/testing/fakecharmstore.go (hunk 4 above)
Comment (frankban) [inline apiserver/testing/fakecharmstore.go]: I agree with you. I'll add a card for abstracting out common code in a separate function. Good point thanks!

**Comment by frankban** [comment]
$$merge$$

**Pull request merged**